src/PluginLoader.php

```diff
@@ -0,0 +1,121 @@
+<?php
+
+declare(strict_types=1);
+
+namespace MyParcelNL\WooCommerce;
+
+use Automattic\WooCommerce\Blocks\Integrations\IntegrationRegistry;
+use MyParcelNL\Pdk\Facade\Installer;
+use MyParcelNL\Pdk\Facade\Pdk;
+use MyParcelNL\WooCommerce\Facade\WooCommerce;
+use MyParcelNL\WooCommerce\Integration\WcBlocksLoader;
+use MyParcelNL\WooCommerce\Service\WordPressHookService;
+
+class PluginLoader
+{
+    public function __construct()
+    {
+        add_action('init', [$this, 'initialize'], 9999);
+
+        // This cannot be done after the init hook
+        add_action('woocommerce_blocks_checkout_block_registration', [$this, 'registerWcCheckoutBlocks']);
+    }
+
+    /**
+     * Perform required tasks that initialize the plugin.
+     *
+     * @throws \Throwable
+     */
+    public function initialize(): void
+    {
+        $this->setup();
+
+        /** @var WordPressHookService $hookService */
+        $hookService = Pdk::get(WordPressHookService::class);
+        $hookService->applyAll();
+    }
+
+    /**
+     * @return void
+     * @throws \MyParcelNL\WooCommerce\Tests\Exception\DieException
```

Comment:
Is het niet erg dat deze Exception uit \Tests\ komt? Of kan de annotatie gewoon weg misschien?

Reply:
die moet weg idd, thanks :)